.github/workflows/build_and_test.yml

```diff
@@ -280,6 +280,7 @@ jobs:
       with:
         name: test-results-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
         path: "**/target/test-reports/*.xml"
+        if-no-files-found: error
```

Comment:
Hm, there's a case when *.xml are not generated time to time although when the tests passed. My only concern is that to make that case also failed after this change.

Reply:
You refer to case `1.` in the description above?

Reply:
Oh no. I meant that there's a case when the tests actually run but XML files are not generated for some reasons I don't know IIRC.

Reply:
I think that is what I meant with `1.`. I have reworded that to be more clear:

> 1. The code has been build […] tests have been executed but no test files have been generated.

This is covered by `if-no-files-found: error` for `actions/upload-artifact`. The upload action is triggered when tests are run, so it expects files to upload.

Reply:
I think this is OK? @HyukjinKwon

Reply:
We can try and go ahead, I'm not against this. My only concern is that we'll now consider the build as failed in the case when the build is successful but it failed to generate XML test reports for an unknown reason - this happens often as far as I remember. Though, I agree that this PR can make such unknown cases onto the surface.